storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java

```diff
@@ -308,6 +308,21 @@ public void updateWeakSubjectivityState(WeakSubjectivityUpdate weakSubjectivityU
     }
   }
 
+  @Override
+  public void storeFinalizedState(BeaconState state) {
+    if (state == null) {
+      return;
+    }
+
+    final Bytes32 blockRoot = spec.getBlockRootAtSlot(state, state.getSlot());
+    try (final FinalizedUpdaterCommon updater = getFinalizedUpdater()) {
+      updater.addFinalizedState(blockRoot, state);
```

Comment:
Worth adding a note that we'll have to come back and also record the finalized state roots as well.  It's a little bit tricky so better to do in a separate PR I think.